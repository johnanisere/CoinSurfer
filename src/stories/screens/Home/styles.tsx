import { StyleSheet } from "react-native";

const styles: any = StyleSheet.create({
  container: {
    backgroundColor: "#FBFAFA"
  },
  row: {
    flex: 1,
    justifyContent: "center"
  },
  text: {
    fontSize: 20,
    marginBottom: 15,
    alignItems: "center"
  },
  mt: {
    marginTop: 18
  },
  leftText: {
    textAlign: "left",
    width: "100%"
  },
  bn: {
    borderBottomWidth: 0
  },
  hoddle: {
    color: "red"
  },
  rise: {
    color: "green"
  }
});
export default styles;
