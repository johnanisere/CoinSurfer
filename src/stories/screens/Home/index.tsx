import * as React from "react";
import { Container, Header, Title, Left, Body, Right } from "native-base";
import Spinner from "./spinner";
import styles from "./styles";
import ListCoin from "./list-coin";
export interface Props {
  list: any;
  fetchCoin: () => void;
  navigation: any;
  fetching: boolean;
  fetched: boolean;
}
export interface State {}
class Home extends React.Component<Props, State> {
  render() {
    const { list, fetching, fetched, fetchCoin } = this.props;

    return (
      <Container style={styles.container}>
        <Header>
          <Left />
          <Body>
            <Title>Coins</Title>
          </Body>
          <Right />
        </Header>
        {fetching && <Spinner />}
        {fetched && (
          <ListCoin list={list} fetching={fetching} fetchCoin={fetchCoin} />
        )}
      </Container>
    );
  }
}

export default Home;
