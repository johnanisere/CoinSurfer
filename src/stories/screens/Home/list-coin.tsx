import * as React from "react";
import { Content, Text, List, ListItem, Body, Left, Right } from "native-base";
import { RefreshControl } from "react-native";
import styles from "./styles";

export interface Props {
  fetchCoin: () => void;
  list: any;
  fetching: boolean;
}
export interface State {}
export default class ListCoins extends React.Component<Props, State> {
  render() {
    const { list, fetching, fetchCoin } = this.props;

    return (
      <Content
        refreshControl={
          <RefreshControl refreshing={fetching} onRefresh={fetchCoin} />
        }
      >
        <List>
          {list.map((item, i) => {
            const price = Number(item.quote.USD.price).toLocaleString();
            const marketCap = Number(item.quote.USD.market_cap).toLocaleString(
              undefined,
              { maximumFractionDigits: 0 }
            );
            const volume_24h = Number(item.quote.USD.volume_24h).toLocaleString(
              undefined,
              { maximumFractionDigits: 0 }
            );
            const percentageChange_1h = `${Number(
              item.quote.USD.percent_change_1h
            ).toFixed(2)}%`;
            const percentageChangeStyle_1h = [
              styles.rise,
              item.quote.USD.percent_change_1h < 0 && styles.hoddle
            ];
            const percentageChange_24h = `${Number(
              item.quote.USD.percent_change_24h
            ).toFixed(2)}%`;
            const percentageChangeStyle_24h = [
              styles.rise,
              item.quote.USD.percent_change_24h < 0 && styles.hoddle
            ];
            const percentageChange_7d = `${Number(
              item.quote.USD.percent_change_7d
            ).toFixed(2)}%`;
            const percentageChangeStyle_7d = [
              styles.rise,
              item.quote.USD.percent_change_7d < 0 && styles.hoddle
            ];
            const time = new Date(
              item.quote.USD.last_updated
            ).toLocaleTimeString();

            return (
              <ListItem key={i} avatar>
                <Left>{/* <Thumbnail source={{ uri: 'Image URL' }} /> */}</Left>
                <Body style={styles.bn}>
                  <Text style={styles.leftText}>
                    Symbol: <Text>{item.symbol}</Text>
                  </Text>
                  <Text style={styles.leftText}>
                    Name: <Text>{item.name}</Text>
                  </Text>
                  <Text style={styles.leftText}>
                    Price: <Text>${price}</Text>
                  </Text>
                  <Text style={styles.leftText}>
                    Market Cap: <Text>${marketCap}</Text>
                  </Text>
                  <Text style={styles.leftText}>
                    Volume(24h): <Text>${volume_24h}</Text>
                  </Text>
                  <Text style={styles.leftText}>
                    Percentage Change(1h):{" "}
                    <Text style={percentageChangeStyle_1h}>
                      {percentageChange_1h}
                    </Text>
                  </Text>
                  <Text style={styles.leftText}>
                    Percentage Change(24h):{" "}
                    <Text style={percentageChangeStyle_24h}>
                      {percentageChange_24h}
                    </Text>
                  </Text>
                  <Text style={styles.leftText}>
                    Percentage Change(7d):{" "}
                    <Text style={percentageChangeStyle_7d}>
                      {percentageChange_7d}
                    </Text>
                  </Text>
                </Body>
                <Right style={styles.bn}>
                  <Text note>last updated</Text>
                  <Text note> {time}</Text>
                </Right>
              </ListItem>
            );
          })}
        </List>
      </Content>
    );
  }
}
