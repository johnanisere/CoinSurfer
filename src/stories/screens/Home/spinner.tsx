import * as React from "react";
import { Container, Spinner } from "native-base";
import styles from "./styles";

export interface Props {}
export interface State {}

class App extends React.Component<Props, State> {
  render() {
    return (
      <Container style={styles.row}>
        <Spinner color="blue" />
      </Container>
    );
  }
}
export default App;
