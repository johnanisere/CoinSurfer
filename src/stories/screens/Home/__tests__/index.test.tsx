import * as React from "react";
import "react-native";
import Home from "../index";
// Note: test renderer must be required after react-native.
import * as renderer from "react-test-renderer";

const navigation = { navigate: jest.fn() };
const list = { map: jest.fn() };
const fetchCoin = jest.fn();

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Home
        list={list}
        fetched={true}
        fetching={false}
        fetchCoin={fetchCoin}
        navigation={navigation}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
