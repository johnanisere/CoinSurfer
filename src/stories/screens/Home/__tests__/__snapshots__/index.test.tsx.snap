// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": "#FBFAFA",
      },
    ]
  }
>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderBottomColor": "#a7a6ab",
          "borderBottomWidth": 0.5,
          "elevation": 3,
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "center",
          "left": 0,
          "paddingLeft": 6,
          "paddingRight": 10,
          "paddingTop": 18,
          "right": 0,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "700",
              "marginLeft": undefined,
              "paddingLeft": 4,
              "paddingTop": 1,
              "textAlign": "center",
            }
          }
        >
          Coins
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      />
    </View>
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": undefined,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[MockFunction]}
        refreshing={false}
      />
    }
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    viewIsInsideTabBar={false}
  >
    <RCTRefreshControl />
    <View>
      <View
        closeOnRowBeginSwipe={false}
        closeOnRowPress={true}
        closeOnScroll={true}
        directionalDistanceChangeThreshold={2}
        disableLeftSwipe={false}
        disableRightSwipe={false}
        leftOpenValue={0}
        previewFirstRow={false}
        recalculateHiddenLayout={false}
        rightOpenValue={0}
        style={Object {}}
        swipeToOpenPercent={50}
      />
    </View>
  </RCTScrollView>
</View>
`;
